lib/discourse_updates.rb

```diff
@@ -115,6 +115,38 @@ def missing_versions
       keys.present? ? keys.map { |k| Discourse.redis.hgetall(k) } : []
     end
 
+    def perform_new_feature_check
+      response = Excon.new(new_features_endpoint).request(expects: [200], method: :Get)
+      json = JSON.parse(response.body)
+      Discourse.redis.set(new_features_key, response.body)
+    end
+
+    def unseen_new_features(user_id)
+      entries = JSON.parse(Discourse.redis.get(new_features_key)) rescue nil
+      return nil if entries.nil?
+
+      last_seen = new_features_last_seen(user_id)
+
+      if last_seen.present?
+        entries.select! { |item| Time.zone.parse(item["created_at"]) > last_seen }
+      end
+
+      entries.sort { |item| Time.zone.parse(item["created_at"]) }
+    end
+
+    def new_features_last_seen(user_id)
```

Comment:
One downside to using redis is we generally consider it impermanent. In theory if redis is cleared, a user will have to acknowledge the same features over and over.

I'm ok to try this out as long as we recognize it might be an issue. If people complain we might want to store it in the database.

Reply:
Yes, I am aware of redis' impermanence but as you note, this will probably be OK. One protection we can add is on the source end of the data to display, we can (and IMO we should) auto-expire items that are 2-3 months old. That would greatly reduce the risk of repeat acknowledgements.